lxd/container_lxc.go

```diff
@@ -6850,6 +6850,10 @@ func (c *containerLXC) insertProxyDevice(devName string, m types.Device) error {
 		return fmt.Errorf("Can't add proxy device to stopped container")
 	}
 
+	if c.tryNat(devName, m) {
```

Comment:
That shouldn't be `try` anymore, it should be that if nat is set to true, we go down the NAT codepath and if that fails, we return an error to the user.